File: ts-sdk/src/sdk/models/operations/renewabranchpassword.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class RenewABranchPasswordPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=branch" })
  branch: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}


export class RenewABranchPasswordRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=read_only_region_id" })
  readOnlyRegionId?: string;
}


export class RenewABranchPasswordRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: RenewABranchPasswordPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: RenewABranchPasswordRequestBody;
}


export class RenewABranchPasswordResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  renewABranchPassword200ApplicationJSONObject?: Record<string, Record<string, any>>;
}

File: go-sdk/pkg/models/operations/renewabranchpassword.go
package operations

type RenewABranchPasswordPathParams struct {
	Branch       string `pathParam:"style=simple,explode=false,name=branch"`
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	ID           string `pathParam:"style=simple,explode=false,name=id"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type RenewABranchPasswordRequestBody struct {
	ReadOnlyRegionID *string `json:"read_only_region_id,omitempty"`
}

type RenewABranchPasswordRequest struct {
	PathParams RenewABranchPasswordPathParams
	Request    *RenewABranchPasswordRequestBody `request:"mediaType=application/json"`
}

type RenewABranchPasswordResponse struct {
	ContentType                                  string
	StatusCode                                   int
	RenewABranchPassword200ApplicationJSONObject map[string]map[string]interface{}
}
